docs/DeveloperGuide.md

```diff
@@ -2,36 +2,530 @@
+
+- `execute()`: This method is responsible for executing the `ListCommand`, listing all drugs in the inventory. It checks
+  the inventory, prepares a user-friendly result message, and returns a `CommandResult` containing the outcome of the
+  command execution, which includes a success message and the list of found `Drug` objects.
+
+### Example Usage
+
+To illustrate how the `ListCommand` works, consider the following example usage:
+
+1. **User Input:** The user initiates the `ListCommand` by entering the following command:
+
+2. **Method Execution:** The `execute()` method within the "ListCommand" class is called.
+
+3. **Inventory Check:** The method checks the inventory to retrieve the list of drugs.
+
+4. **Result Building:** If the inventory contains drugs, the method constructs a success message (e.g., "Listed all
+   drugs in the inventory.") and includes the list of drugs with relevant details.
+
+5. **User Feedback:** The result is displayed to the user, showing a clear and organized list of drugs in the inventory.
+
+The "ListCommand" enhances the user's ability to access inventory information efficiently and is designed to handle
+various inventory sizes while providing a user-friendly experience.
+
+The following sequence diagram shows how the Find Command function works.
+
+<img src="UML Diagrams/ListCommandDiagram.png" width="350">
+
+---
+
+## 3. Delete Command
+
+The "Delete" function is designed to enable users to remove specific drugs from the inventory based on the drug's name, 
+to remove drugs they no longer need, fully depleted or discontinued.
+
+**Design Considerations**
+
+1. **User-Specified Drug Name:** The function allows users to specify the drug name they want to delete from the
+   inventory.
+
+2. **Data Integrity:** It ensures that the deletion operation maintains the integrity of the inventory data structure,
+   updating it correctly.
+
+**Implementation and Rationale**
+
+This method is executed when the delete command is invoked. First, it attempts to delete a drug from the inventory 
+using the inventory.deleteDrug(this.keyToDelete) method. If successful, it retrieves the deleted entry.
+If the drug is successfully deleted, it returns a success message. 
+If the drug is not found (i.e., a DrugNotFoundException is thrown), it returns a failure message.
+
+---
+
+## 4. Help Command
+
+The command is responsible for showing users a list of all possible commands.
+
+**Design Considerations**
+
+The command was designed to print out and show how to use a list of all possible commands in a neat and concise way.
+
+**Implementation and Rationale**
+
+Command will use java's system out to print out all required information with a blank line in between for clarity.
+
+---
+
+## 5. Save Command
+
+The save command was made as a means to backup user entered drug data into the hard drive of the computer to ensure
+previously entered data is saved and accessable whenever the app is launched.
```

Comment:
Could you tell when would you execute the save method? Whenever any update is made or when the user exit the app?